### PR TITLE
TransactionSignatureChecker bug fix and tests

### PR DESCRIPTION
**Pull request by benthecarman**

### CODE REVIEW

File: core/src/main/scala/org/bitcoins/core/crypto/TransactionSignatureChecker.scala [hunk 1]
```diff
@@ -18,6 +20,32 @@ import scala.annotation.tailrec
...
+    checkSignature(txSignatureComponent, PartialSignature(pubKey, signature))
+
+  def checkSignature(
+      txSignatureComponent: TxSigComponent,
+      partialSignature: PartialSignature): TransactionSignatureCheckerResult = {
+    checkSignature(txSignatureComponent,
+                   txSignatureComponent.output.scriptPubKey.asm.toList,
+                   partialSignature.pubKey,
+                   partialSignature.signature,
+                   Policy.standardFlags)
```
Comment (nkohen) [inline core/src/main/scala/org/bitcoins/core/crypto/TransactionSignatureChecker.scala]: If this is the default value might as well leave it off

File: core/src/main/scala/org/bitcoins/core/crypto/TransactionSignatureChecker.scala [hunk 2]
```diff
@@ -18,6 +20,32 @@ import scala.annotation.tailrec
...
+
+  def checkSignature(
+      txSignatureComponent: TxSigComponent,
+      script: Seq[ScriptToken],
+      partialSignature: PartialSignature): TransactionSignatureCheckerResult =
+    checkSignature(txSignatureComponent,
+                   script,
+                   partialSignature.pubKey,
+                   partialSignature.signature,
+                   Policy.standardFlags)
```
Comment (nkohen) [inline core/src/main/scala/org/bitcoins/core/crypto/TransactionSignatureChecker.scala]: ditto

File: core-test/src/test/scala/org/bitcoins/core/crypto/TransactionSignatureCheckerTest.scala [hunk 3]
```diff
@@ -1,8 +1,229 @@
...
 
 /**
   * Created by chris on 2/29/16.
   */
-class TransactionSignatureCheckerTest extends BitcoinSUnitTest {}
+class TransactionSignatureCheckerTest extends BitcoinSUnitTest {
+
+  behavior of "TransactionSignatureChecker"
+
+  it must "validate a P2PK sig" in {
```
Comment (nkohen) [inline core-test/src/test/scala/org/bitcoins/core/crypto/TransactionSignatureCheckerTest.scala:36]: for all of these validation tests you should add invalidation tests or asserts for if something small is changed

**Review by nkohen — APPROVED** [review]
ACK assuming CI passes

**Review by Christewart — APPROVED** [review]

**Pull request merged**